nixos/modules/services/misc/jackett.nix

```diff
@@ -58,6 +64,18 @@ in
       description = "Jackett";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
+      preStart = # bash
+        ''
+          set -euo pipefail
+          umask 077 # ServerConfig.json should be 700
+
+          config_file="${cfg.dataDir}/ServerConfig.json"
+          generated="{}"
+          [ -f "$config_file" ] && generated=$(cat "$config_file")
+
+          nixcfg='${builtins.toJSON cfg.serverConfig}'
```

Comment:
This is not safe as the JSON may contain `'`. Either encode it properly (there are helpers in nixpkgs) or better yet is to put this into an environment variable or file so that it can be read with no encoding required.